components/popover/popover-mixin.js

```diff
@@ -124,20 +127,26 @@ export const PopoverMixin = superclass => class extends superclass {
 	}
 
 	close() {
-		this._opened = false;
-		return this.updateComplete;
+		if (!this._opened) return Promise.resolve();
+		return new Promise(resolve => {
+			this.addEventListener('d2l-popover-close', resolve, { once: true });
+			this._opened = false;
+		});
 	}
 
 	configure(properties) {
 		this._noAutoClose = properties?.noAutoClose ?? false;
 		this._noAutoFocus = properties?.noAutoFocus ?? false;
-		this._opened = properties?.opened ?? false;
+		this._trapFocus = properties?.trapFocus ?? false;
 	}
 
 	open(applyFocus = true) {
-		this._applyFocus = applyFocus !== undefined ? applyFocus : true;
-		this._opened = true;
-		return this.updateComplete;
+		if (this._opened) return Promise.resolve();
+		return new Promise(resolve => {
```

Comment:
This sugar turned out to be useful in some of the tests.

Reply:
One concern I have is that the open (and also close) related functionality is quite split up - some is here and then some is in `updated`. If that's how it has to happen because of rendering then that's fair enough, just thought I'd raise it.

Reply:
That's a good point. Now that the consumer is required to use the `open()`, `close()`, `toggleOpen()` methods, I think we should be able to move that logic so that it's not split.

Reply:
Ok... I think I have that tidied up.